### ISSUE TITLE
Option to throw raw exceptions from client pipeline

### ISSUE THREAD

**Issue by WaffleLapkin** [ktorio/ktor#1101]
### Ktor Version
1.1.5

### Ktor Engine Used(client or server and name)
client, issue doesn't depend on client type.

### JVM Version, Operating System and Relevant Context
also doesn't matter.

### Feedback
At the moment any exception thrown by client pipeline ([except](https://github.com/ktorio/ktor/blob/1.1.x/ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt#L58-L62) `BadResponseStatusException`) will be wraped in `ReceivePipelineException`.

I consider there must be an option to throw 'raw' exception. 
E.g. it can be done in that way (I don't really sure if it is clean API):
```kotlin
class RawPipelineException(val cause: Throwable) : Exception()
```
and adding
```kotlin
catch (e: RawPipelineException) {
    throw e.cause
}
```
to [HttpClientCall.kt#L60](https://github.com/ktorio/ktor/blob/1.1.x/ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt#L60)

### Motivation
In my case, I have written validation in `HttpClientFeature` and want to throw validation exceptions.

For now, I'm using the following workaround:
```kotlin
/**
 * Executes a [HttpClient] POST request, with the specified [scheme], [host], [port], [path] and [body].
 * And allows to further configure the request, using a [block] receiving an [HttpRequestBuilder].
 *
 * Tries to receive a specific type [T], if fails, an exception is thrown.
 *
 * Also unwraps [ReceivePipelineException] if [ReceivePipelineException.cause] is [TelegramAPIException]
 */
suspend inline fun <reified T> HttpClient.telegramPost(
    scheme: String = "http", host: String = "localhost", port: Int = DEFAULT_PORT,
    path: String = "/",
    body: Any = EmptyContent,
   block: HttpRequestBuilder.() -> Unit = {}
): T = try {
    post(scheme, host, port, path, body, block)
} catch (e: ReceivePipelineException) {
    if (e.cause is TelegramAPIException) throw e.cause
    else throw e
}
```

But it isn't clean way, I think.

**Comment by e5l** [comment]
Hi @WaffleLapkin, it should be fixed in `1.2.0-rc`. Could you check it?

**Comment by WaffleLapkin** [comment]
In `1.2.0-rc` this work as expected, thanks.
